journal/recorder.cpp

```diff
@@ -34,6 +36,19 @@ void Recorder::WriteAtDestruction(serialization::Method const& method) {
 void Recorder::Activate(base::not_null<Recorder*> const recorder) {
   CHECK(active_recorder_ == nullptr);
   active_recorder_ = recorder;
+
+  // When the recorder gets activated, pretend that we got a GetVersion call.
```

Comment:
Why pretend? Couldn’t we just call `principia__GetVersion` here instead of reimplementing it?

We might have to move it to its own interface_something file for cleaner includes, but splitting the main `interface` file tends to be something we want to do anyway…

Reply:
I started along that path and it's not nice.  The API of the interface is not exactly designed for being called from C++.  In this case the parameters are `char const** const`, which would require awkward storage management.  It's much simpler to just set two fields of a proto.